polyfill/lib/regex.mjs

```diff
@@ -1,18 +1,7 @@
-const tzComponent = /\.[-A-Za-z_]|\.\.[-A-Za-z._]{1,12}|\.[-A-Za-z_][-A-Za-z._]{0,12}|[A-Za-z_][-A-Za-z._]{0,13}/;
-const offsetIdentifierNoCapture = /(?:[+\u2212-][0-2][0-9](?::?[0-5][0-9])?)/;
+const offsetIdentifierNoCapture = /(?:[+\u2212-](?:[01][0-9]|2[0-3])(?::?[0-5][0-9])?)/;
+const tzComponent = /[A-Za-z._][A-Za-z._0-9+-]*/;
```

Comment:
Do any corresponding changes need to be made in validStrings.mjs?

Reply:
I think for practical reasons we currently don't reflect the grammar for time zone and calendar identifiers in the validStrings test. It'd be vanishingly unlikely that a valid one would be generated randomly.

Reply:
validStrings.mjs bottoms out named time zones in `Intl.supportedValuesOf('timeZone')`: https://github.com/tc39/proposal-temporal/blob/e09bf4f7774cd8ae94dbe243b7356fc03589218f/polyfill/test/validStrings.mjs#L10

And since this PR relaxes the grammar, all of those continue to pass. We could assert specifically that IANA-invalid names like "Foo/Bar+1" are _accepted_ by the grammar, but I don't think there's much value in that.